### ISSUE TITLE
APIv2 /containers/{id}/wait doesn't work with correct condition enumeration

### ISSUE THREAD

**Issue by matejvasek** [containers/podman#8697]
Currently the used enumeration is here:
podman/libpod/define/containerstate.go

```go
const (
	ContainerStateUnknown ContainerStatus = iota
	ContainerStateConfigured ContainerStatus = iota
	ContainerStateCreated ContainerStatus = iota
	ContainerStateRunning ContainerStatus = iota
	ContainerStateStopped ContainerStatus = iota
	ContainerStatePaused ContainerStatus = iota
	ContainerStateExited ContainerStatus = iota
	ContainerStateRemoving ContainerStatus = iota
)
```

But docker uses different values:
```go
const (
	WaitConditionNotRunning WaitCondition = "not-running"
	WaitConditionNextExit   WaitCondition = "next-exit"
	WaitConditionRemoved    WaitCondition = "removed"
)
```

**Comment by matejvasek** [comment]
Calls like this fails.
```
POST /v1.38/containers/ctrId/wait?condition=next-exit
```

**Comment by matejvasek** [comment]
I could work on this if nobody else wants to.

**Comment by rhatdan** [comment]
You got it.

**Comment by rhatdan** [comment]
@matejvasek Any progress?

**Comment by matejvasek** [comment]
@rhatdan I should get to this soon. I think I could get PR sometimes next week.

Unfortunately, before starting working directly on this I spent (or wasted?) some time on related issue. 
I really didn't like the idea of checking state of a container every 250ns. I was trying do add some kind of notification mechanism to `lock.Locker`, something like condition variable. Long story short: it didn't work.

There is a problem when waiting process is forcefully killed, like with `SIGKILL` (however `SIGTERM` or `SIGINT` can be handled).
First I tried to use standard posix cond var. The issues is that posix cond var is not robust.
Then I tried to write my own cond var using semaphores,
but there would be a small memory leak anytime waiting process dies.
Last but not least, this would work only with `SHM` lock, not `FS` locks.
